packages/rpc-types/src/cluster-url.ts

```diff
@@ -0,0 +1,14 @@
+export type MainnetUrl = string & { '~cluster': 'mainnet' };
+export type DevnetUrl = string & { '~cluster': 'devnet' };
+export type TestnetUrl = string & { '~cluster': 'testnet' };
+export type ClusterUrl = string | MainnetUrl | DevnetUrl | TestnetUrl;
+
+export function mainnet(putativeString: string): MainnetUrl {
```

Comment:
Just a nit regarding my favorite word to use in code. This is usually the word I use for _assertion_ functions (ie. ones whose return type is `foo is Bar`. Putative means ‘supposed’ or ‘assumed to exist.’

This puppy doesn't make any assertions; it's just a typecast utility. `mainnet(url: string): MainnetUrl` would do fine.

Reply:
This is your favorite word in code? What about `foo`? Personally, mine is `console.log("FLARE")`, but that never gets committed :).